modules/text/include/opencv2/text/swtTextDetection.hpp

```diff
@@ -0,0 +1,21 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef SWT_H
+#define SWT_H
+
+#include <opencv2/core.hpp>
+#include <vector>
+
+namespace cv {
+namespace text {
+/** @brief Applies the Stroke Width Transform operator followed by filtering of connected components of similar Stroke Widths to find letter candidates and chain them by proximity and size.
+    @param input_image the input image with 3 channels.
```

Comment:
the actual algorithm works on single channel grayscale  (and the multi-channel image is used , to draw visualizations into it) right ?

while being very useful, imho those visualizations  should not be mandatory.

what about this idea: pass an optional drawing image as another argument, like:

    void SWTTextDetection (InputArray input_image, bool dark_on_light, OutputArray draw=noArray()) {
        if (draw.needed()) {
                // if they passed in the color image, use it,
                // if they passed in an empty Mat, create a 3chan grayscale image and draw into that
        }
    }

Reply:
> the actual algorithm works on single channel grayscale

Most of it yes, there is a part where the color averages are taken into account. The paper assumes two consecutive letter candidates to have a similar color average. So color image is pretty much mandatory.

> pass an optional drawing image as another argument

Awesome, I will do that. thanks!

Reply:
> (the findValidChains function)

ah, right. missed it.